catroid/src/androidTest/java/org/catrobat/catroid/test/common/DeviceVariableAccessorTest.java

```diff
@@ -0,0 +1,180 @@
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.io.asynctask.ProjectCopyTask;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.ui.recyclerview.controller.SpriteController;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(AndroidJUnit4.class)
+public class DeviceVariableAccessorTest {
+
+	UserVariable userVariable;
+	UserVariable localUserVariable;
+	Sprite sprite;
+	Project project;
+	Context context;
+	ArrayList<String> cleanupList = new ArrayList<>();
+
+	@Before
+	public void setUp() throws IOException {
+		context = InstrumentationRegistry.getTargetContext();
+		project = createSimpleProject("projectX");
+		sprite = new Sprite("_sprite_");
+		project.getDefaultScene().addSprite(sprite);
+		userVariable = new UserVariable("X", 10.0);
+		localUserVariable = new UserVariable("Y", 20.0);
+
+		ProjectManager.getInstance().setCurrentProject(project);
+		sprite.addUserVariable(localUserVariable);
+		project.addUserVariable(userVariable);
+		XstreamSerializer.getInstance().saveProject(project);
+		cleanupList.add(project.getName());
+	}
+
+	@Test
+	public void saveUserVariableTest() throws IOException {
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+
+		Object savedVar = userVariable.getValue();
+		Object localSavedVar = localUserVariable.getValue();
+		accessor.writeVariable(userVariable);
+		accessor.writeVariable(localUserVariable);
+		userVariable.setValue("false");
+		localUserVariable.setValue("false");
+		assertTrue(accessor.readUserVariableValue(userVariable));
+		assertTrue(accessor.readUserVariableValue(localUserVariable));
+		assertEquals(savedVar, userVariable.getValue());
+		assertEquals(localSavedVar, localUserVariable.getValue());
+	}
+
+	@Test
+	public void saveProjectTest() throws IOException, LoadingProjectException {
+		UserVariable userVariable1 = project.getUserVariable(userVariable.getName());
+		userVariable1.setValue("false");
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(project.getDirectory());
+		accessor.writeVariable(userVariable1);
+		XstreamSerializer.getInstance().saveProject(project);
+		Project project1 = XstreamSerializer.getInstance().loadProject(project.getDirectory(), context);
+
+		UserVariable result = project1.getUserVariable(userVariable.getName());
+		accessor = new DeviceVariableAccessor(project1.getDirectory());
+		accessor.readUserVariableValue(result);
+
+		assertEquals(result.getDeviceValueKey(), userVariable1.getDeviceValueKey());
+		assertEquals(result.getValue(), userVariable1.getValue());
+	}
+
+	@Test
+	public void copyProjectTest() throws IOException, LoadingProjectException {
```

Comment:
is this necessary in this form?
there should be one test that asserts that the DeviceVariable file is copied with the rest of the project, and that test should not be in this file but somewhere over in the IO package where all the other tests regarding IO operations are.
no loading or saving using the accessor required.